sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -452,7 +453,14 @@ class RelationalGroupedDataset protected[sql](
       case RelationalGroupedDataset.GroupByType =>
         val valueExprs = values.map(_ match {
           case c: Column => c.expr
-          case v => Literal.apply(v)
+          case v =>
+            try {
+              Literal.apply(v)
+            } catch {
+              case _: SparkRuntimeException =>
+                throw QueryExecutionErrors.pivotColumnUnsupportedError(
+                  v, pivotColumn.expr.dataType)
```

Comment:
This can fit to one line (100 chars):
```suggestion
                throw QueryExecutionErrors.pivotColumnUnsupportedError(v, pivotColumn.expr.dataType)
```

Reply:
thanks for fixing it!